### DISCUSSION TITLE
Regular Expressions --regex

### DISCUSSION THREAD

**Discussion by oPromessa** [RhetTbull/osxphotos#523] [Q&A] [answered]
Awesome tool! Many Kudos @RhetTbull 

I'm moving from iPhoto into Photos with about 80k referenced files... Photos is not able to  _find/locate_ several referenced files while it does others. But I digress! Hope I can get out on the other side. ;)

I'm testing out osxphotos to regularly update an export folder from Photos. One (maybe) issue and one question came up:

1. I don't seem to to get `--regex` to work to select multiple albums. From the command below ii only selects pics from the *last mentioned* regex, that is "1992-08-12" album folder. Am I doing something wrong?
   - `--regex "2021-03-14 Home" "{folder_album}" --regex "2021-04-09" "{folder_album}" --regex "2019-07-12" "{folder_album}" --regex "1992-08-12" "{folder_album}" `
   -  The Readme does mention:
      -  > "You may specify more than one regular expression match by repeating '--regex' with different arguments."

2. What regular expressions does the `--regex` parameter accept (the full spectrum)? I was trying out thinks like `"2020-[01][09]"` and other more complex options but they do not seem to work!
   - [ ] **UPDATE:** User error! Was able to make it work with one single `regex` like this `--regex "2021-03-14 Home|2021-04-09|1992-08-12|2019-07-12" "{folder_album}"`

Again, excellent tool. Thanks!

**Comment by RhetTbull** [comment]
Glad you find osxphotos useful!

--regex should work as used in your first example. It accepts any valid python regex expression. I'll take a look at the code to see why only the last pattern was matching.

**Reply by RhetTbull** [reply]
Yep, it's a bug...I'll open an issue and get it fixed. The results were being re-initialized between regexs which explains why only the last was working:

https://github.com/RhetTbull/osxphotos/blob/4a81e643a7edba6bdea028607d6134197dc67b1e/osxphotos/photosdb/photosdb.py#L3301

**Comment by RhetTbull** [comment, marked as answer]
@oPromessa I can confirm you found a bug.  It's now fixed in [v0.42.92](https://github.com/RhetTbull/osxphotos/releases/tag/v0.42.92) so you can now use multiple `--regex` values and get the expected behavior.

Regarding your question on which regex syntax can be used, any valid [python regex](https://docs.python.org/3/library/re.html) should work.  You can specify case-insensitivity using the `-i, --ignore-case` flag.  The example you gave `"2020-[01][09]"` may not do what you think it's doing.  This says match `2020-` followed by any character in the set `{0,1}` then any character in the set `{0,9}` so the following would match: `2020-00`, `2020-09`, `2020-10`, `2020-19`.
